spec/js/post_office_search_spec.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { getByLabelText, getByRole, waitFor } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+const wait = (ms: number = 0) =>
+  new Promise<void>((res) => {
+    setTimeout(() => {
+      res();
+    }, ms);
+  });
+
+describe('Post Office Search', () => {
+  // This is simulated since the test env doesn't support fetch
+  const testServerPort = 9835;
+  const addressSearchUrl = `http://127.0.0.1:${testServerPort}/api/address_search`;
+  const locationsSearchUrl = `http://127.0.0.1:${testServerPort}/api/usps_locations`;
+
+  beforeEach(() => {
+    if ('fetch' in global) {
+      jest.spyOn(global, 'fetch');
+    } else {
+      // @ts-ignore
+      global.fetch = jest.fn();
+    }
+    (global.fetch as jest.Mock).mockImplementation(() => ({
+      get ok() {
+        return true;
+      },
+      json: jest.fn(() => []),
+      get headers() {
+        return new Headers();
+      },
+    }));
+  });
+
+  let container: HTMLDivElement;
+  let user: ReturnType<typeof userEvent['setup']>;
+  beforeEach(async () => {
+    user = userEvent.setup();
+    await act(async () => {
+      container = document.createElement('div');
+      container.id = 'post-office-search';
+      container.dataset.addressSearchUrl = addressSearchUrl;
+      container.dataset.locationsSearchUrl = locationsSearchUrl;
+      document.body.appendChild(container);
+      jest.isolateModules(() => {
+        // eslint-disable-next-line global-require
+        require('../../assets/js/post_office_search');
+      });
+      await wait();
+    });
+  });
+
+  afterEach(() => {
+    if (container) {
+      act(() => {
+        container.remove();
+      });
+    }
+  });
```

Comment:
Is this trying to reset the states between test cases? Could we just `page.reload()`?

Reply:
This particular test is working with `jsdom`, not puppeteer. [`jsdom` doesn't support navigation](https://github.com/jsdom/jsdom#unimplemented-parts-of-the-web-platform).